File: src/modelo/Producao.java
package modelo;

import java.util.ArrayList;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;

import dao.IDInterface;

@Entity
public class Producao implements IDInterface{
	
	@Id
	@GeneratedValue (strategy=GenerationType.IDENTITY)
	private int id;
	private String data;
	@ManyToOne(cascade=CascadeType.ALL)
	private Prato prato;
	@ManyToOne(cascade=CascadeType.ALL)
	private Funcionario cozinheiro;
	@OneToMany(mappedBy="produto",cascade=CascadeType.ALL,orphanRemoval=true)
	private List<Avaliacao> avaliacoes = new ArrayList <Avaliacao> ();
	
	public Producao () {
		
	}
	
	public Producao(String data, Prato prato, Funcionario cozinheiro) {
		super();
		this.data = data;
		this.prato = prato;
		this.cozinheiro = cozinheiro;
	}
		
	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	public String getData() {
		return data;
	}
	public void setData(String data) {
		this.data = data;
	}
	public Prato getPrato() {
		return prato;
	}
	public void setPrato(Prato prato) {
		this.prato = prato;
	}
	public Funcionario getCozinheiro() {
		return cozinheiro;
	}
	public void setCozinheiro(Funcionario cozinheiro) {
		this.cozinheiro = cozinheiro;
	}
	public List<Avaliacao> getAvaliacoes() {
		return avaliacoes;
	}
	public void setAvaliacoes(List<Avaliacao> avaliacoes) {
		this.avaliacoes = avaliacoes;
	}
	@Override
	public String toString() {
		return "Producao [data=" + data + ", prato=" + prato + ", cozinheiro=" + cozinheiro + ", avaliacoes="
				+ avaliacoes + "]";
	}
	
	
	

}

File: src/modelo/Producao.php
<?php
namespace projeto\modelo;
use \projeto\DAO\IDInterface;
// use modelo\Prato;
// use modelo\Funcionario;

class Producao implements IDInterface {
  private $data;
  private $prato;
  private $cozinheiro;
  private $avaliacoes;

  public function __construct (String $data,Prato $prato, Funcionario $cozinheiro ) {
    $this->data=$data;
    $this->prato=$prato;
    $this->cozinheiro = $cozinheiro;
    $this->avaliacoes= [];
  }

  public function getId () : int {
    return $this->id;
  }

  public function setId (int $id) {
    $this->id = $id;
  }

  public function getData () : String {
    return $this->data;
  }

  public function setData (String $data) {
    $this->data = $data;
  }

  public function getPrato () : Prato {
    return $this->prato;
  }

  public function setPrato (Prato $prato) {
    $this->prato = $prato;
  }

  public function getCozinheiro () : Funcionario {
    return $this->cozinheiro;
  }

  public function setCozinheiro (Funcionario $cozinheiro) {
    $this->cozinheiro = $cozinheiro;
  }

  public function getAvaliacoes () : array {
    return $this->avaliacoes;
  }

  public function setAvaliacoes (array $avaliacoes) {
    $this->avaliacoes = $avaliacoes;
  }

  public function __toString()
  {
    return 'Data: '.$this->data.'/Prato: '.$this->prato->getNome().'/Cozinheiro: '.$this->cozinheiro->getNome ().'/Quant avaliações: '.count($this->avaliacoes);
  }

}

 ?>
